src/test/java/software/amazon/documentdb/jdbc/query/limitations/DocumentDbSqlInjectionTest.java

```diff
@@ -0,0 +1,301 @@
+        final String query2 = String.format(
+                "SELECT \"%1$s\", \"%2$s\", \"%3$s\" FROM (SELECT * FROM \"%4$s\".\"%5$s\"" +
+                        " WHERE \"%1$s\" = '%6$s')",
+                primaryKeyColumnName,
+                "field",
+                "field1",
+                getDatabaseName(),
+                COLLECTION_NAME + "_array",
+                injection);
+        final DocumentDbMqlQueryContext queryContext2 = queryMapper.get(query2);
+        Assertions.assertNotNull(queryContext2);
+        final List<Bson> aggregateOperations2 = queryContext2.getAggregateOperations();
+        assertKeyNotExists(expectedKey, aggregateOperations2);
+        assertValueExists(injection, aggregateOperations2);
+
+        final String query3 = String.format(
+                "SELECT \"%1$s\", \"%2$s\", \"%3$s\" FROM \"%4$s\".\"%5$s\"" +
+                        " WHERE \"%1$s\" = SUBSTRING('%6$s', 1, 2000)",
+                primaryKeyColumnName,
+                "field",
+                "field1",
+                getDatabaseName(),
+                COLLECTION_NAME + "_array",
+                injection);
+        final DocumentDbMqlQueryContext queryContext3 = queryMapper.get(query3);
+        Assertions.assertNotNull(queryContext3);
+        final List<Bson> aggregateOperations3 = queryContext3.getAggregateOperations();
+        assertKeyNotExists(expectedKey, aggregateOperations3);
+        assertValueExists(injection, aggregateOperations3);
+
+        final String query4 = String.format(
+                "SELECT \"%1$s\", \"%2$s\", \"%3$s\" FROM \"%4$s\".\"%5$s\"" +
+                        " WHERE \"%1$s\" = CONCAT('%6$s', '')",
+                primaryKeyColumnName,
+                "field",
+                "field1",
+                getDatabaseName(),
+                COLLECTION_NAME + "_array",
+                injection);
+        final DocumentDbMqlQueryContext queryContext4 = queryMapper.get(query4);
+        Assertions.assertNotNull(queryContext4);
+        final List<Bson> aggregateOperations4 = queryContext4.getAggregateOperations();
+        assertKeyNotExists(expectedKey, aggregateOperations4);
+        assertValueExists(injection, aggregateOperations4);
+
+        final String query5 = String.format(
+                "SELECT \"%1$s\", \"%2$s\", \"%3$s\" FROM \"%4$s\".\"%5$s\"" +
+                        " WHERE \"%1$s\" = REVERSE('%6$s')",
+                primaryKeyColumnName,
+                "field",
+                "field1",
+                getDatabaseName(),
+                COLLECTION_NAME + "_array",
+                new StringBuilder(injection).reverse());
+        final DocumentDbMqlQueryContext queryContext5 = queryMapper.get(query5);
+        Assertions.assertNotNull(queryContext5);
+        final List<Bson> aggregateOperations5 = queryContext5.getAggregateOperations();
+        assertKeyNotExists(expectedKey, aggregateOperations5);
+        assertValueExists(injection, aggregateOperations5);
```

Comment:
I'm confused here, if we're making sure Mongo injection is not possible, then why are we making sure that the injection exists in the aggregate options (i.e., by calling `assertValueExists(injection, <aggregateOptionsVar> )`)?

Reply:
@alinaliBQ 
The point of this assertion is that the injection has been interpreted only as a value, as is. We are ensuring the (attempted) injection despite quoting will only be interpreted as a value and nothing else.

Reply:
Oh I see, thanks for explaining!